Isn't work in termux

asdf install python 3.9.18
Downloading python-build...
Cloning into '/data/data/com.termux/files/home/.asdf/plugins/python/pyenv'...
remote: Enumerating objects: 23384, done.
remote: Counting objects: 100% (1834/1834), done.
remote: Compressing objects: 100% (176/176), done.
remote: Total 23384 (delta 1722), reused 1700 (delta 1652), pack-reused 21550
Receiving objects: 100% (23384/23384), 4.66 MiB | 2.70 MiB/s, done.
Resolving deltas: 100% (15893/15893), done.
python-build 3.9.18 /data/data/com.termux/files/home/.asdf/installs/python/3.9.18
Downloading Python-3.9.18.tar.xz...
-> https://www.python.org/ftp/python/3.9.18/Python-3.9.18.tar.xz
Installing Python-3.9.18...

BUILD FAILED (Linux 4.14.113-24323734 using python-build 2.3.25-6-g91002438)

Inspect or clean up the working tree at /data/data/com.termux/files/usr/tmp/python-build.20230904211745.3387
Results logged to /data/data/com.termux/files/usr/tmp/python-build.20230904211745.3387.log

Last 10 log lines:
3 warnings generated.
gcc -c -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -DOPENSSL_NO_SSL3   -std=c99 -Wextra -Wno-unused-result -Wno-unused-parameter -Wno-missing-field-initializers -Werror=implicit-function-declaration -fvisibility=hidden  -I./Include/internal  -I. -I./Include -I/data/data/com.termux/files/home/.asdf/installs/python/3.9.18/include -I/data/data/com.termux/files/home/.asdf/installs/python/3.9.18/include -fPIC -DPy_BUILD_CORE -o Python/traceback.o Python/traceback.c
gcc -c -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -DOPENSSL_NO_SSL3   -std=c99 -Wextra -Wno-unused-result -Wno-unused-parameter -Wno-missing-field-initializers -Werror=implicit-function-declaration -fvisibility=hidden  -I./Include/internal  -I. -I./Include -I/data/data/com.termux/files/home/.asdf/installs/python/3.9.18/include -I/data/data/com.termux/files/home/.asdf/installs/python/3.9.18/include -fPIC -DPy_BUILD_CORE -o Python/getopt.o Python/getopt.c
In file included from Python/thread.c:80:
Python/thread_pthread.h:473:33: error: call to undeclared function 'sem_clockwait'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
            status = fix_status(sem_clockwait(thelock, CLOCK_MONOTONIC,
                                ^
1 error generated.
make: *** [Makefile:1803: Python/thread.o] Error 1
make: *** Waiting for unfinished jobs....

--- a/asdf.nu
+++ b/asdf.nu
@@ -1,14 +1,27 @@
 def-env configure-asdf [] {
+    $env.ASDF_DIR = (
+      if ($env | get --ignore-errors ASDF_NU_DIR | is-empty) == false {
+        $env.ASDF_NU_DIR
+      }
+      else if ($env | get --ignore-errors ASDF_DIR | is-empty) == false {
+        $env.ASDF_DIR
+      } else {
+        print --stderr "asdf: Either ASDF_NU_DIR or ASDF_DIR must not be empty"
+        return
+      }
+    )
 
-    $env.ASDF_DIR = ( if ( $env | get --ignore-errors ASDF_DIR | is-empty ) { $env.ASDF_NU_DIR } else { $env.ASDF_DIR } )
-
-    let shims_dir = ( if ( $env | get --ignore-errors ASDF_DATA_DIR | is-empty ) { $env.HOME | path join '.asdf' } else { $env.ASDF_DIR } | path join 'shims' )
-
+    let shims_dir = (
+      if ( $env | get --ignore-errors ASDF_DATA_DIR | is-empty ) {
+        $env.HOME | path join '.asdf'
+      } else {
+        $env.ASDF_DIR
+      } | path join 'shims'
+    )
     let asdf_bin_dir = ( $env.ASDF_DIR | path join 'bin' )
 
     $env.PATH = ( $env.PATH | split row (char esep) | where { |p| $p != $shims_dir } | prepend $shims_dir )
     $env.PATH = ( $env.PATH | split row (char esep) | where { |p| $p != $asdf_bin_dir } | prepend $asdf_bin_dir )
-
 }
 
 configure-asdf

--- a/docs/guide/getting-started.md
+++ b/docs/guide/getting-started.md
@@ -300,7 +300,7 @@ Add the following to `~/.config/powershell/profile.ps1`:
 Add `asdf.nu` to your `~/.config/nushell/config.nu` with:
 
 ```shell
-"\n$env.ASDF_NU_DIR = ($env.HOME | path join '.asdf')\n source " + ($env.HOME | path join '.asdf/asdf.nu') | save --append $nu.config-path
+"\n$env.ASDF_DIR = ($env.HOME | path join '.asdf')\n source " + ($env.HOME | path join '.asdf/asdf.nu') | save --append $nu.config-path
 ```
 
 Completions are automatically configured
@@ -311,7 +311,7 @@ Completions are automatically configured
 Add `asdf.nu` to your `~/.config/nushell/config.nu` with:
 
 ```shell
-"\n$env.ASDF_NU_DIR = (brew --prefix asdf | str trim | into string | path join 'libexec')\n source " +  (brew --prefix asdf | into string | path join 'libexec/asdf.nu') | save --append $nu.config-path
+"\n$env.ASDF_DIR = (brew --prefix asdf | str trim | into string | path join 'libexec')\n source " +  (brew --prefix asdf | into string | path join 'libexec/asdf.nu') | save --append $nu.config-path
 ```
 
 Completions are automatically configured
@@ -322,7 +322,7 @@ Completions are automatically configured
 Add `asdf.nu` to your `~/.config/nushell/config.nu` with:
 
 ```shell
-"\n$env.ASDF_NU_DIR = '/opt/asdf-vm/'\n source /opt/asdf-vm/asdf.nu" | save --append $nu.config-path
+"\n$env.ASDF_DIR = '/opt/asdf-vm/'\n source /opt/asdf-vm/asdf.nu" | save --append $nu.config-path
 ```
 
 Completions are automatically configured.

--- a/docs/guide/getting-started.md
+++ b/docs/guide/getting-started.md
@@ -311,7 +311,7 @@ Completions are automatically configured
 Add `asdf.nu` to your `~/.config/nushell/config.nu` with:
 
 ```shell
-"\n$env.ASDF_DIR = (brew --prefix asdf | str trim | into string | path join 'libexec')\n source " +  (brew --prefix asdf | into string | path join 'libexec/asdf.nu') | save --append $nu.config-path
+"\n$env.ASDF_DIR = (brew --prefix asdf | str trim | into string | path join 'libexec')\n source " +  (brew --prefix asdf | str trim | into string | path join 'libexec/asdf.nu') | save --append $nu.config-path
 ```
 
 Completions are automatically configured

--- a/docs/pt-br/guide/getting-started.md
+++ b/docs/pt-br/guide/getting-started.md
@@ -308,7 +308,7 @@ Ao concluir atualizará automaticamente
 Adicione `asdf.nu` ao seu `~/.config/nushell/config.nu` através do comando:
 
 ```shell
-"\n$env.ASDF_DIR = (brew --prefix asdf | str trim | into string | path join 'libexec')\n source " +  (brew --prefix asdf | into string | path join 'libexec/asdf.nu') | save --append $nu.config-path
+"\n$env.ASDF_DIR = (brew --prefix asdf | str trim | into string | path join 'libexec')\n source " +  (brew --prefix asdf | str trim | into string | path join 'libexec/asdf.nu') | save --append $nu.config-path
 ```
 
 Ao concluir atualizará automaticamente

--- a/docs/pt-br/guide/getting-started.md
+++ b/docs/pt-br/guide/getting-started.md
@@ -297,7 +297,7 @@ Adicione a seguinte linha ao seu `~/.config/powershell/profile.ps1`:
 Adicione `asdf.nu` ao seu `~/.config/nushell/config.nu` através do comando:
 
 ```shell
-"\n$env.ASDF_NU_DIR = ($env.HOME | path join '.asdf')\n source " + ($env.HOME | path join '.asdf/asdf.nu') | save --append $nu.config-path
+"\n$env.ASDF_DIR = ($env.HOME | path join '.asdf')\n source " + ($env.HOME | path join '.asdf/asdf.nu') | save --append $nu.config-path
 ```
 
 Ao concluir atualizará automaticamente
@@ -308,7 +308,7 @@ Ao concluir atualizará automaticamente
 Adicione `asdf.nu` ao seu `~/.config/nushell/config.nu` através do comando:
 
 ```shell
-"\n$env.ASDF_NU_DIR = (brew --prefix asdf | str trim | into string | path join 'libexec')\n source " +  (brew --prefix asdf | into string | path join 'libexec/asdf.nu') | save --append $nu.config-path
+"\n$env.ASDF_DIR = (brew --prefix asdf | str trim | into string | path join 'libexec')\n source " +  (brew --prefix asdf | into string | path join 'libexec/asdf.nu') | save --append $nu.config-path
 ```
 
 Ao concluir atualizará automaticamente
@@ -319,7 +319,7 @@ Ao concluir atualizará automaticamente
 Adicione `asdf.nu` ao seu `~/.config/nushell/config.nu` através do comando:
 
 ```shell
-"\n$env.ASDF_NU_DIR = '/opt/asdf-vm/'\n source /opt/asdf-vm/asdf.nu" | save --append $nu.config-path
+"\n$env.ASDF_DIR = '/opt/asdf-vm/'\n source /opt/asdf-vm/asdf.nu" | save --append $nu.config-path
 ```
 
 Ao concluir atualizará automaticamente

--- a/docs/zh-hans/guide/getting-started.md
+++ b/docs/zh-hans/guide/getting-started.md
@@ -309,7 +309,7 @@ echo -e "\n. \"$(brew --prefix asdf)/libexec/asdf.ps1\"" >> ~/.config/powershell
 使用以下命令将 `asdf.nu` 加入到 `~/.config/nushell/config.nu` 文件中:
 
 ```shell
-"\n$env.ASDF_DIR = (brew --prefix asdf | str trim | into string | path join 'libexec')\n source " +  (brew --prefix asdf | into string | path join 'libexec/asdf.nu') | save --append $nu.config-path
+"\n$env.ASDF_DIR = (brew --prefix asdf | str trim | into string | path join 'libexec')\n source " +  (brew --prefix asdf | str trim | into string | path join 'libexec/asdf.nu') | save --append $nu.config-path
 ```
 
 补全功能将会自动配置。

--- a/docs/zh-hans/guide/getting-started.md
+++ b/docs/zh-hans/guide/getting-started.md
@@ -298,7 +298,7 @@ echo -e "\n. \"$(brew --prefix asdf)/libexec/asdf.ps1\"" >> ~/.config/powershell
 使用以下命令将 `asdf.nu` 加入到 `~/.config/nushell/config.nu` 文件中：
 
 ```shell
-"\n$env.ASDF_NU_DIR = ($env.HOME | path join '.asdf')\n source " + ($env.HOME | path join '.asdf/asdf.nu') | save --append $nu.config-path
+"\n$env.ASDF_DIR = ($env.HOME | path join '.asdf')\n source " + ($env.HOME | path join '.asdf/asdf.nu') | save --append $nu.config-path
 ```
 
 补全功能将会自动配置。
@@ -309,7 +309,7 @@ echo -e "\n. \"$(brew --prefix asdf)/libexec/asdf.ps1\"" >> ~/.config/powershell
 使用以下命令将 `asdf.nu` 加入到 `~/.config/nushell/config.nu` 文件中:
 
 ```shell
-"\n$env.ASDF_NU_DIR = (brew --prefix asdf | str trim | into string | path join 'libexec')\n source " +  (brew --prefix asdf | into string | path join 'libexec/asdf.nu') | save --append $nu.config-path
+"\n$env.ASDF_DIR = (brew --prefix asdf | str trim | into string | path join 'libexec')\n source " +  (brew --prefix asdf | into string | path join 'libexec/asdf.nu') | save --append $nu.config-path
 ```
 
 补全功能将会自动配置。
@@ -320,7 +320,7 @@ echo -e "\n. \"$(brew --prefix asdf)/libexec/asdf.ps1\"" >> ~/.config/powershell
 使用以下命令将 `asdf.nu` 加入到 `~/.config/nushell/config.nu` 文件中:
 
 ```shell
-"\n$env.ASDF_NU_DIR = '/opt/asdf-vm/'\n source /opt/asdf-vm/asdf.nu" | save --append $nu.config-path
+"\n$env.ASDF_DIR = '/opt/asdf-vm/'\n source /opt/asdf-vm/asdf.nu" | save --append $nu.config-path
 ```
 
 补全功能将会自动配置。

--- a/test/asdf_nu.bats
+++ b/test/asdf_nu.bats
@@ -26,7 +26,7 @@ run_nushell() {
     hide-env -i asdf
     hide-env -i ASDF_DIR
     \$env.PATH = ( '$(cleaned_path)' | split row ':' )
-    \$env.ASDF_NU_DIR = '$PWD'
+    \$env.ASDF_DIR = '$PWD'
 
     source asdf.nu
     $1"
@@ -60,12 +60,27 @@ run_nushell() {
   [ "$result" = "" ]
 }
 
-@test "retains ASDF_DIR" {
+@test "retains ASDF_DIR (from ASDF_NU_DIR)" {
   run nu -c "
     hide-env -i asdf
     \$env.ASDF_DIR = ( pwd )
     \$env.PATH = ( '$(cleaned_path)' | split row ':' )
     \$env.ASDF_NU_DIR = '$PWD'
+
+    source asdf.nu
+
+    echo \$env.ASDF_DIR"
+
+  [ "$status" -eq 0 ]
+  [ "$output" = "$PWD" ]
+}
+
+@test "retains ASDF_DIR (from ASDF_DIR)" {
+  run nu -c "
+    hide-env -i asdf
+    \$env.ASDF_DIR = ( pwd )
+    \$env.PATH = ( '$(cleaned_path)' | split row ':' )
+    \$env.ASDF_DIR = '$PWD'
 
     source asdf.nu
 

